src/main/java/org/springframework/data/domain/Range.java

```diff
@@ -87,11 +196,216 @@ public boolean contains(T value) {
 
 		Assert.notNull(value, "Reference value must not be null!");
 
-		boolean greaterThanLowerBound = lowerBound == null ? true
-				: lowerInclusive ? lowerBound.compareTo(value) <= 0 : lowerBound.compareTo(value) < 0;
-		boolean lessThanUpperBound = upperBound == null ? true
-				: upperInclusive ? upperBound.compareTo(value) >= 0 : upperBound.compareTo(value) > 0;
+		boolean greaterThanLowerBound = lowerBound.getValue() //
+				.map(it -> lowerBound.isInclusive() ? it.compareTo(value) <= 0 : it.compareTo(value) < 0) //
+				.orElse(true);
+
+		boolean lessThanUpperBound = upperBound.getValue() //
+				.map(it -> upperBound.isInclusive() ? it.compareTo(value) >= 0 : it.compareTo(value) > 0) //
+				.orElse(true);
 
 		return greaterThanLowerBound && lessThanUpperBound;
 	}
+
+	/* 
+	 * (non-Javadoc)
+	 * @see java.lang.Object#toString()
+	 */
+	@Override
+	public String toString() {
+		return String.format("%s-%s", lowerBound.toPrefixString(), upperBound.toSuffixString());
+	}
+
+	/**
+	 * Value object representing a boundary. A boundary can either be {@link #unbounded() unbounded},
+	 * {@link #inclusive(Comparable) including its value} or {@link #exclusive(Comparable) its value}.
+	 * 
+	 * @author Mark Paluch
+	 * @since 2.0
+	 * @soundtrack Mohamed Ragab - Excelsior Sessions (March 2017)
+	 */
+	@Value
```

Comment:
I guess we need to add an `@RequiredArgsConstructor(access = AccessLevel.PRIVATE)` to make sure the default constructor is not public.